pyci/rdm/algorithms.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np 
+from abc import ABCMeta, abstractmethod
+class Projection(metaclass=ABCMeta):
+    
+    @abstractmethod
+    def __init__(self, initial_guess,constraints):
+        self.initial_guess = initial_guess
+        self.constraints = constraints
+    
+    @abstractmethod 
+    def optimize(self):
+        pass
+
+class Dykstra(metaclass=ABCMeta):
+    def __init__(self, initial_guess:float, constraints:list, alpha:int =1, max_iterations:int =100, eps:int =1e-6):
```

Comment:
eps: has to be float and alpha should be float
You can safely remove type check in here, because you provide default value